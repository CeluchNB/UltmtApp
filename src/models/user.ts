import { Realm } from '@realm/react'

export const DisplayUserSchema: Realm.ObjectSchema = {
    name: 'DisplayUser',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string',
        localGuest: 'bool?',
    },
}
