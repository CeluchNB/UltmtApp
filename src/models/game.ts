import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { Tournament } from '../types/tournament'
import { CreateGame, Game, GameStatus, PointStats } from '../types/game'
import { DisplayTeam, GuestTeam } from '../types/team'

export class GameSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Game',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            creator: 'DisplayUser',
            teamOne: 'DisplayTeam',
            teamTwo: 'GuestTeam',
            teamTwoDefined: 'bool',
            scoreLimit: 'int',
            halfScore: 'int',
            startTime: 'date',
            softcapMins: 'int',
            hardcapMins: 'int',
            playersPerPoint: 'int',
            timeoutPerHalf: 'int',
            floaterTimeout: 'bool',
            tournament: 'Tournament?',
            teamOneScore: 'int',
            teamTwoScore: 'int',
            teamOnePlayers: 'DisplayUser[]',
            teamTwoPlayers: 'DisplayUser[]',
            resolveCode: 'string',
            statsPoints: 'PointStats[]',
            offline: 'bool',
            teamOneStatus: 'string',
            teamTwoStatus: 'string',
        },
    }

    _id: string
    creator: DisplayUser
    teamOne: DisplayTeam
    teamTwo: GuestTeam
    teamTwoDefined: boolean
    scoreLimit: number
    halfScore: number
    startTime: Date
    softcapMins: number
    hardcapMins: number
    playersPerPoint: number
    timeoutPerHalf: number
    floaterTimeout: boolean
    tournament?: Tournament
    teamOneScore: number
    teamTwoScore: number
    teamOnePlayers: DisplayUser[]
    teamTwoPlayers: DisplayUser[]
    resolveCode: string
    statsPoints: PointStats[]
    offline: boolean
    teamOneStatus: GameStatus
    teamTwoStatus: GameStatus

    constructor(
        game: Game,
        offline: boolean = false,
        statsPoints: PointStats[] = [],
    ) {
        this._id = game._id
        this.creator = game.creator
        this.teamOne = game.teamOne
        this.teamTwo = game.teamTwo
        this.teamTwoDefined = game.teamTwoDefined
        this.scoreLimit = game.scoreLimit
        this.halfScore = game.halfScore
        this.startTime = game.startTime
        this.softcapMins = game.softcapMins
        this.hardcapMins = game.hardcapMins
        this.playersPerPoint = game.playersPerPoint
        this.timeoutPerHalf = game.timeoutPerHalf
        this.floaterTimeout = game.floaterTimeout
        this.tournament = game.tournament
        this.teamOneScore = game.teamOneScore
        this.teamTwoScore = game.teamTwoScore
        this.teamOnePlayers = game.teamOnePlayers
        this.teamTwoPlayers = game.teamTwoPlayers
        this.resolveCode = game.resolveCode || ''
        this.statsPoints = statsPoints
        this.offline = offline
        this.teamOneStatus = game.teamOneStatus
        this.teamTwoStatus = game.teamTwoStatus
    }

    static createOfflineGame(game: CreateGame, teamOnePlayers: DisplayUser[]) {
        return {
            _id: new Realm.BSON.ObjectID().toHexString(),
            creator: game.creator,
            teamOne: game.teamOne,
            teamTwo: game.teamTwo,
            teamTwoDefined: game.teamTwoDefined,
            scoreLimit: game.scoreLimit,
            halfScore: game.halfScore,
            startTime: new Date(game.startTime),
            softcapMins: game.softcapMins,
            hardcapMins: game.hardcapMins,
            playersPerPoint: game.playersPerPoint,
            timeoutPerHalf: game.timeoutPerHalf,
            floaterTimeout: game.floaterTimeout,
            tournament: game.tournament,
            teamOneScore: 0,
            teamTwoScore: 0,
            teamOneActive: true,
            teamTwoActive: false,
            teamOnePlayers: teamOnePlayers,
            teamTwoPlayers: [],
            resolveCode: '',
            statsPoints: [],
            offline: true,
        }
    }
}
export const PointStatsSchema: Realm.ObjectSchema = {
    name: 'PointStats',
    embedded: true,
    properties: {
        _id: 'string',
        pointStats: 'PlayerPointStats[]',
    },
}

export const PlayerPointStatsSchema: Realm.ObjectSchema = {
    name: 'PlayerPointStats',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string?',
        pointsPlayed: 'int',
        goals: 'int',
        assists: 'int',
        turnovers: 'int',
        blocks: 'int',
    },
}
