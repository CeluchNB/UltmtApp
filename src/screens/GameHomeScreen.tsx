import { Game } from '../types/game'
import GameCard from '../components/atoms/GameCard'
import MapSection from '../components/molecules/MapSection'
import React from 'react'
import { TextInput } from 'react-native-paper'
import { searchGames } from '../services/data/game'
import { SafeAreaView, ScrollView, StyleSheet } from 'react-native'
import { useColors, useData } from '../hooks'

const GameHomeScreen: React.FC<{}> = () => {
    const { colors } = useColors()

    const { data: liveGames, loading: liveLoading } = useData<Game[]>(
        searchGames,
        undefined,
        true,
    )
    const { data: recentGames, loading: recentLoading } = useData<Game[]>(
        searchGames,
        undefined,
        false,
    )

    const renderGame = (game: Game) => {
        return (
            <GameCard
                key={game._id}
                teamOne={game.teamOne}
                teamTwo={game.teamTwo}
                teamOneScore={game.teamOneScore}
                teamTwoScore={game.teamTwoScore}
                scoreLimit={game.scoreLimit}
            />
        )
    }

    const styles = StyleSheet.create({
        screen: {
            height: '100%',
            backgroundColor: colors.primary,
        },
        title: { alignSelf: 'center' },
        input: {
            backgroundColor: colors.primary,
            color: colors.textPrimary,
            width: '90%',
            alignSelf: 'center',
            marginBottom: 5,
            fontSize: 20,
        },
        container: {
            width: '90%',
            alignSelf: 'center',
        },
    })

    return (
        <SafeAreaView style={styles.screen}>
            <TextInput
                mode="flat"
                style={[styles.input]}
                underlineColor={colors.textPrimary}
                activeUnderlineColor={colors.textPrimary}
                placeholderTextColor={colors.gray}
                theme={{
                    colors: {
                        text: colors.textPrimary,
                    },
                }}
                placeholder={`Search games...`}
            />
            <ScrollView style={styles.container}>
                <MapSection
                    title="Live Games"
                    showButton={false}
                    showCreateButton={false}
                    listData={liveGames}
                    renderItem={renderGame}
                    loading={liveLoading}
                />
                {recentGames && recentGames.length > 0 && (
                    <MapSection
                        title="Recent Games"
                        showButton={true}
                        showCreateButton={false}
                        listData={recentGames}
                        renderItem={renderGame}
                        loading={recentLoading}
                        buttonText="explore more games"
                    />
                )}
            </ScrollView>
        </SafeAreaView>
    )
}

export default GameHomeScreen
