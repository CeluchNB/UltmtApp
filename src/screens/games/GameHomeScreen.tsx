import { FAB } from 'react-native-paper'
import { Game } from '../../types/game'
import GameCard from '../../components/atoms/GameCard'
import { GameHomeProps } from '../../types/navigation'
import MapSection from '../../components/molecules/MapSection'
import SearchBar from '../../components/atoms/SearchBar'
import { searchGames } from '../../services/data/game'
import React, { useMemo } from 'react'
import {
    RefreshControl,
    SafeAreaView,
    ScrollView,
    StyleSheet,
    Text,
} from 'react-native'
import { useData, useTheme } from '../../hooks'

const GameHomeScreen: React.FC<GameHomeProps> = ({ navigation }) => {
    const {
        theme: { colors, size },
    } = useTheme()

    const { data, loading, refetch } = useData<Game[]>(searchGames)

    const navigateToSearch = (live: string) => {
        navigation.navigate('GameSearch', { live })
    }

    const navigateToCreate = () => {
        navigation.navigate('GameCreationFlow', { screen: 'SelectMyTeam' })
    }

    const navigateToViewGame = (gameId: string) => {
        navigation.navigate('ViewGame', { gameId })
    }

    const renderGame = (game: Game) => {
        return (
            <GameCard
                key={game._id}
                game={game}
                onPress={() => {
                    navigateToViewGame(game._id)
                }}
            />
        )
    }

    const liveGames = useMemo(() => {
        return data?.filter(g => g.teamOneActive)
    }, [data])

    const recentGames = useMemo(() => {
        return data?.filter(g => !g.teamOneActive)
    }, [data])

    const styles = StyleSheet.create({
        screen: {
            height: '100%',
            backgroundColor: colors.primary,
        },
        title: { alignSelf: 'center' },
        container: {
            width: '90%',
            alignSelf: 'center',
        },
        errorText: {
            alignSelf: 'center',
            fontSize: size.fontThirty,
            color: colors.gray,
        },
        fab: {
            position: 'absolute',
            margin: 20,
            right: 0,
            bottom: 0,
            borderRadius: 8,
            backgroundColor: colors.textPrimary,
        },
    })

    return (
        <SafeAreaView style={styles.screen}>
            <SearchBar
                placeholder="Search games..."
                onPress={() => navigateToSearch('undefined')}
            />
            <ScrollView
                style={styles.container}
                testID="game-home-scroll-view"
                showsVerticalScrollIndicator={false}
                refreshControl={
                    <RefreshControl
                        onRefresh={refetch}
                        refreshing={loading}
                        colors={[colors.textSecondary]}
                    />
                }>
                {liveGames && liveGames.length > 0 && (
                    <MapSection
                        title="Live Games"
                        showButton={true}
                        showCreateButton={false}
                        listData={liveGames}
                        renderItem={renderGame}
                        loading={loading}
                        buttonText="explore live games"
                        onButtonPress={() => {
                            navigateToSearch('true')
                        }}
                    />
                )}
                {recentGames && recentGames.length > 0 && (
                    <MapSection
                        title="Recent Games"
                        showButton={true}
                        showCreateButton={false}
                        listData={recentGames}
                        renderItem={renderGame}
                        loading={false}
                        buttonText="explore recent games"
                        onButtonPress={() => {
                            navigateToSearch('false')
                        }}
                    />
                )}
                {(!data || data?.length < 1) && (
                    <Text style={styles.errorText}>
                        No games available currently. Try refreshing or
                        searching.
                    </Text>
                )}
            </ScrollView>
            <FAB icon="plus" style={styles.fab} onPress={navigateToCreate} />
        </SafeAreaView>
    )
}

export default GameHomeScreen
