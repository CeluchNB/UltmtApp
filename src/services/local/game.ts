import * as Constants from '../../utils/constants'
import AsyncStorage from '@react-native-async-storage/async-storage'
import { DisplayUser } from '../../types/user'
import { getRealm } from '../../models/realm'
import { throwApiError } from '../../utils/service-utils'
import { ActionSchema, GameSchema, PointSchema } from '../../models'
import { CreateGame, Game } from '../../types/game'

const parseGame = (schema: GameSchema): Game & { offline: boolean } => {
    return JSON.parse(
        JSON.stringify({
            _id: schema._id,
            creator: schema.creator,
            teamOne: schema.teamOne,
            teamTwo: schema.teamTwo,
            teamTwoDefined: schema.teamTwoDefined,
            scoreLimit: schema.scoreLimit,
            halfScore: schema.halfScore,
            startTime: schema.startTime,
            softcapMins: schema.softcapMins,
            hardcapMins: schema.hardcapMins,
            playersPerPoint: schema.playersPerPoint,
            timeoutPerHalf: schema.timeoutPerHalf,
            floaterTimeout: schema.floaterTimeout,
            tournament: schema.tournament,
            teamOneScore: schema.teamOneScore,
            teamTwoScore: schema.teamTwoScore,
            teamOneActive: schema.teamOneActive,
            teamTwoActive: schema.teamTwoActive,
            teamOnePlayers: schema.teamOnePlayers,
            teamTwoPlayers: schema.teamTwoPlayers,
            resolveCode: schema.resolveCode,
            points: schema.points,
            offline: schema.offline,
        }),
    )
}

/**
 * Method to determine if currently active game is offline
 * @returns boolean indicating current game is offline or not
 */
export const activeGameOffline = async (): Promise<boolean> => {
    return (await AsyncStorage.getItem('active_game_offline')) === 'true'
}

/**
 * Method to get the ObjectId of the current active game
 * @returns current active game id
 */
export const activeGameId = async (): Promise<string> => {
    return (await AsyncStorage.getItem('active_game_id')) || ''
}

/**
 * Method to set active game offline status
 * @param offline boolean
 */
export const setActiveGameOffline = async (offline: boolean) => {
    await AsyncStorage.setItem('active_game_offline', offline.toString())
}

/**
 * Method to set active game id
 * @param id string
 */
export const setActiveGameId = async (id: string) => {
    await AsyncStorage.setItem('active_game_id', id)
}

export const createOfflineGame = async (
    data: CreateGame,
    teamOnePlayers: DisplayUser[],
): Promise<string> => {
    const realm = await getRealm()

    let id: string = ''
    realm.write(() => {
        const game = realm.create<GameSchema>(
            'Game',
            GameSchema.createOfflineGame(data, teamOnePlayers),
        )
        id = game._id
    })

    return id
}

export const saveGame = async (game: Game) => {
    const realm = await getRealm()
    const gameRecord = await realm.objectForPrimaryKey<GameSchema>(
        'Game',
        game._id,
    )
    let offline = false
    if (gameRecord) {
        offline = gameRecord.offline
    }
    const schema = new GameSchema(game, offline)
    realm.write(() => {
        realm.create('Game', schema, Realm.UpdateMode.Modified)
    })
}

export const getGameById = async (
    gameId: string,
): Promise<Game & { offline: boolean }> => {
    const realm = await getRealm()
    const game = await realm.objectForPrimaryKey<GameSchema>('Game', gameId)
    if (!game) {
        return throwApiError({}, Constants.GET_GAME_ERROR)
    }
    return parseGame(game)
}

export const activeGames = async (
    userId: string,
): Promise<(Game & { offline: boolean })[]> => {
    const realm = await getRealm()
    const games = await realm.objects<GameSchema>('Game')

    return games
        .filter(g => g.creator._id === userId)
        .map(game => parseGame(game))
}

export const deleteFullGame = async (gameId: string): Promise<void> => {
    const realm = await getRealm()
    const game = await realm.objectForPrimaryKey<GameSchema>('Game', gameId)

    const points = game?.points.map(id => {
        return realm.objectForPrimaryKey<PointSchema>('Point', id)
    })

    const actions: (ActionSchema | null)[] = []
    points?.forEach(p => {
        p?.teamOneActions.forEach(id => {
            const objectId = new Realm.BSON.ObjectID(id)
            actions.push(
                realm.objectForPrimaryKey<ActionSchema>('Action', objectId),
            )
        })
        p?.teamTwoActions.forEach(id => {
            const objectId = new Realm.BSON.ObjectID(id)
            actions.push(
                realm.objectForPrimaryKey<ActionSchema>('Action', objectId),
            )
        })
    })

    realm.write(() => {
        realm.delete(actions.filter(a => a !== null))
        realm.delete(points?.filter(p => p !== null))
        realm.delete(game)
    })
}
