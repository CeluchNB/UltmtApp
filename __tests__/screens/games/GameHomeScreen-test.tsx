import * as GameServices from '../../../src/services/data/game'
import * as UserServices from '../../../src/services/data/user'
import GameHomeScreen from '../../../src/screens/games/GameHomeScreen'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import React from 'react'
import { render } from '@testing-library/react-native'
import store from '../../../src/store/store'
import { waitUntilRefreshComplete } from '../../../fixtures/utils'
import { QueryClient, QueryClientProvider } from 'react-query'
import { fetchProfileData, game } from '../../../fixtures/data'

jest.mock('../../../src/components/atoms/GameCard', () => () => {
    return <div>Game</div>
})

const props = {
    navigation: {
        navigate: jest.fn(),
        addListener: jest.fn(),
    } as any,
    route: {} as any,
}

const client = new QueryClient()

describe('GameHomeScreen', () => {
    beforeAll(() => {
        jest.useFakeTimers({ legacyFakeTimers: true })
    })

    afterAll(() => {
        jest.useRealTimers()
    })

    beforeEach(() => {
        jest.spyOn(GameServices, 'searchGames').mockReturnValueOnce(
            Promise.resolve([
                game,
                {
                    ...game,
                    _id: 'game2',
                    teamTwo: { name: 'Team 7' },
                    teamTwoScore: 7,
                },
            ]),
        )

        jest.spyOn(UserServices, 'fetchProfile').mockReturnValueOnce(
            Promise.resolve(fetchProfileData),
        )
    })

    it('should match snapshot with live and recent data', async () => {
        const snapshot = render(
            <Provider store={store}>
                <QueryClientProvider client={client}>
                    <NavigationContainer>
                        <GameHomeScreen {...props} />
                    </NavigationContainer>
                </QueryClientProvider>
            </Provider>,
        )

        await waitUntilRefreshComplete(
            snapshot.getByTestId('game-home-scroll-view'),
        )

        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('should match snapshot with no data', async () => {
        jest.spyOn(GameServices, 'searchGames')
            .mockReset()
            .mockReturnValueOnce(Promise.resolve([]))

        const snapshot = render(
            <Provider store={store}>
                <QueryClientProvider client={client}>
                    <NavigationContainer>
                        <GameHomeScreen {...props} />
                    </NavigationContainer>
                </QueryClientProvider>
            </Provider>,
        )

        await waitUntilRefreshComplete(
            snapshot.getByTestId('game-home-scroll-view'),
        )

        expect(snapshot.toJSON()).toMatchSnapshot()
    })
})
