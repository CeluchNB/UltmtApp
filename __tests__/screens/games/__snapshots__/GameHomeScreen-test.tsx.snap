// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with live and recent data 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "display": "flex",
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "backgroundColor": "#303030",
            "color": "#ffd600",
            "flex": 1,
            "marginBottom": 5,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#ffd600",
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": [
              {
                "scaleY": 0.5,
              },
            ],
            "zIndex": 1,
          }
        }
        testID="text-input-underline"
      />
      <View
        style={
          [
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "minHeight": 56,
            },
          ]
        }
      >
        <TextInput
          editable={true}
          focusable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onPressOut={[Function]}
          placeholder="Search games..."
          placeholderTextColor="#CEC6B4"
          selectionColor="#ffd600"
          style={
            [
              {
                "margin": 0,
              },
              {
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "#ffd600",
                "fontFamily": "System",
                "fontSize": 41.0958904109589,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
              undefined,
            ]
          }
          testID="search-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        colors={
          [
            "#7AC6FF",
          ]
        }
        onRefresh={[Function]}
        refreshing={false}
        tintColor="#7AC6FF"
      />
    }
    showsVerticalScrollIndicator={false}
    style={
      {
        "alignSelf": "center",
        "width": "90%",
      }
    }
    testID="game-home-scroll-view"
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          {
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "#ffd600",
                "flex": 1,
                "fontSize": 61.64383561643835,
                "fontWeight": "bold",
                "marginBottom": 10,
              }
            }
          >
            Live Games
          </Text>
        </View>
        <div>
          Game
        </div>
        <div>
          Game
        </div>
        <View
          collapsable={false}
          style={
            {
              "alignSelf": "center",
              "bottom": undefined,
              "end": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "start": undefined,
              "top": undefined,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "marginTop": 10,
                  "minWidth": 64,
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    false,
                    {
                      "borderRadius": 20,
                    },
                  ]
                }
                testID="more-button"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        {
                          "textAlign": "left",
                        },
                        {
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "marginHorizontal": 16,
                            "marginVertical": 9,
                            "textAlign": "center",
                          },
                          false,
                          {
                            "marginHorizontal": 12,
                          },
                          undefined,
                          {
                            "textTransform": "uppercase",
                          },
                          {
                            "color": "#7AC6FF",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    explore live games
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": 0,
        "end": undefined,
        "left": undefined,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 8,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 3,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        style={
          {
            "backgroundColor": "#ffd600",
            "borderRadius": 8,
            "margin": 20,
            "opacity": 1,
            "overflow": "hidden",
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
        testID="fab-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              undefined,
            ]
          }
          testID="fab"
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 16,
                  "height": 56,
                  "width": 56,
                },
              ]
            }
            testID="fab-content"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#303030",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰐕
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`should match snapshot with no data 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "display": "flex",
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgba(231, 224, 236, 1)",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "backgroundColor": "#303030",
            "color": "#ffd600",
            "flex": 1,
            "marginBottom": 5,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#ffd600",
            "bottom": 0,
            "height": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": [
              {
                "scaleY": 0.5,
              },
            ],
            "zIndex": 1,
          }
        }
        testID="text-input-underline"
      />
      <View
        style={
          [
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "minHeight": 56,
            },
          ]
        }
      >
        <TextInput
          editable={true}
          focusable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onPressOut={[Function]}
          placeholder="Search games..."
          placeholderTextColor="#CEC6B4"
          selectionColor="#ffd600"
          style={
            [
              {
                "margin": 0,
              },
              {
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              {
                "height": 56,
              },
              {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              {
                "color": "#ffd600",
                "fontFamily": "System",
                "fontSize": 41.0958904109589,
                "fontWeight": undefined,
                "letterSpacing": 0.15,
                "lineHeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              [
                {},
              ],
              undefined,
            ]
          }
          testID="search-input"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <RCTScrollView
    refreshControl={
      <RefreshControlMock
        colors={
          [
            "#7AC6FF",
          ]
        }
        onRefresh={[Function]}
        refreshing={false}
        tintColor="#7AC6FF"
      />
    }
    showsVerticalScrollIndicator={false}
    style={
      {
        "alignSelf": "center",
        "width": "90%",
      }
    }
    testID="game-home-scroll-view"
  >
    <RCTRefreshControl />
    <View>
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#CEC6B4",
            "fontSize": 61.64383561643835,
          }
        }
      >
        No games available currently. Try refreshing or searching.
      </Text>
    </View>
  </RCTScrollView>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": 0,
        "end": undefined,
        "left": undefined,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 8,
        "start": undefined,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 3,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        style={
          {
            "backgroundColor": "#ffd600",
            "borderRadius": 8,
            "margin": 20,
            "opacity": 1,
            "overflow": "hidden",
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
        testID="fab-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              undefined,
            ]
          }
          testID="fab"
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 16,
                  "height": 56,
                  "width": 56,
                },
              ]
            }
            testID="fab-content"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#303030",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰐕
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
